tests/data/test_huggingface_dataset_and_source.py

```diff
@@ -100,8 +100,9 @@ def test_from_huggingface_dataset_constructs_expected_dataset_with_data_dir(tmp_
     df = pd.DataFrame.from_dict({"a": [1, 2, 3], "b": [4, 5, 6]})
     data_dir = "data"
     os.makedirs(tmp_path / data_dir)
-    df.to_csv(tmp_path / data_dir / "my_data.csv")
-    ds = datasets.load_dataset(str(tmp_path), data_dir=data_dir, name="csv", split="train")
+    csv_path = tmp_path / data_dir / "my_data.csv"
+    df.to_csv(csv_path)
+    ds = datasets.load_dataset("csv", data_files=str(csv_path), split="train")
```

Comment:
Used the same approach as the official doc:

https://huggingface.co/docs/datasets/loading#local-and-remote-files

```python
from datasets import load_dataset
dataset = load_dataset("csv", data_files="my_file.csv")
```